src/equations/shallow_water_wet_dry_1d.jl

```diff
@@ -382,170 +400,174 @@ Further details on this hydrostatic reconstruction and its motivation can be fou
+                                                                                equations.H0))
 end
 
 # Specialized `FluxHLL` to avoid spurious dissipation in the bottom topography
-@inline function (numflux::Trixi.FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
-                                          equations::ShallowWaterEquationsWetDry1D)
-    return (numflux::Trixi.FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
-                                    Trixi.ShallowWaterEquations1D(gravity_constant = equations.gravity))
+@inline function (numflux::FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
+                                    equations::ShallowWaterEquationsWetDry1D)
+    return (numflux::FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
+                              ShallowWaterEquations1D(equations.gravity, equations.H0))
 end
 
-# TODO: This function is currently exported by Trixi.jl. Needs to be uncommented when removed from Trixi.jl
-# """
-#     min_max_speed_chen_noelle(u_ll, u_rr, orientation::Integer,
-#                               equations::ShallowWaterEquationsWetDry1D)
+"""
+    min_max_speed_chen_noelle(u_ll, u_rr, orientation::Integer,
+                              equations::ShallowWaterEquations1D)
 
-# The approximated speeds for the HLL type numerical flux used by Chen and Noelle for their
-# hydrostatic reconstruction. As they state in the paper, these speeds are chosen for the numerical
-# flux to ensure positivity and to satisfy an entropy inequality.
+The approximated speeds for the HLL type numerical flux used by Chen and Noelle for their
+hydrostatic reconstruction. As they state in the paper, these speeds are chosen for the numerical
+flux to ensure positivity and to satisfy an entropy inequality.
 
-# Further details on this hydrostatic reconstruction and its motivation can be found in
-# - Guoxian Chen and Sebastian Noelle (2017)
-#   A new hydrostatic reconstruction scheme based on subcell reconstructions
-#   [DOI:10.1137/15M1053074](https://dx.doi.org/10.1137/15M1053074)
-# """
-# @inline function min_max_speed_chen_noelle(u_ll, u_rr, orientation::Integer,
-#                                            equations::ShallowWaterEquationsWetDry1D)
-#     # Get the velocity quantities
-#     v_ll = velocity(u_ll, equations)
-#     v_rr = velocity(u_rr, equations)
+Further details on this hydrostatic reconstruction and its motivation can be found in
+- Guoxian Chen and Sebastian Noelle (2017)
+  A new hydrostatic reconstruction scheme based on subcell reconstructions
+  [DOI:10.1137/15M1053074](https://dx.doi.org/10.1137/15M1053074)
+"""
+# Since FluxHLL is dispatched on ShallowWaterEquations1D this is only used by the ShallowWaterEquations1D.
+@inline function min_max_speed_chen_noelle(u_ll, u_rr, orientation::Integer,
+                                           equations::ShallowWaterEquations1D)
```

Comment:
Why does this not dispatch on `ShallowWaterEquationsWetDry1D`? As far as I recall, all of the Chen & Noelle functionality was moved out of the standard `ShallowWaterEquations` from Trixi.

Reply:
Yes, we have moved all the Chen & Noelle related functions out of Trixi. This has gotten a bit confusing, since we still have some functionality in Trixi.

Currently, this wavespeed estimate is only ever used by the `flux_hll_chen_noelle`.
Now when we call `flux_hll_chen_noelle` the `FluxHll` is dispatched on the standard `ShallowWaterEquations` from Trixi.jl, because this still contains the shallow water specific `FluxHLL` that ignores the bottom topography. So the `min_max_speed_chen_noelle` is then only called from this `FluxHLL` of the `ShallowWaterEquations`.

Reply:
Interesting, and a bit odd, but I suppose it is nothing a front facing user would need to worry about. We should maybe add a comment above this function explaining this.

One follow-up question. If I want to add a new wave speed estimate for use with `FluxHLL` would I always need to have it dispatch on Trixi's `ShallowWaterEquations` then?

Reply:
Yes this would always need to dispatch on the `ShallowWaterEquations`. 
While this doesn't affect the users, the way this is distributed between `Trixi.jl `and `TrixiShallowWater.jl` is not obvious and very confusing for developers.
Instead in 0fc33bc6d0b7a51cdacc81dd936bd37830e161a1 the `ShallowWaterEquationsWetDry` now has its own `FluxHLL`, together with wrapper functions for the wave speed estimates. This introduces some additional code, but I believe it is much clearer how everything works.

Reply:
Yes, this introduces some extra wrapper functions, but I think it will lead to less confusion from us in the future.

Reply:
Would it make sense to add a layer `AbstractShallowWaterEquations{NDIMS}` for that purpose? Or would that be too broad (it might apply to too many equations where it is not valid)?

Reply:
@sloede do you mean introducing a single function that would dispatch on `AbstractShallowWaterEquations{NDIMS}` instead?
```
@inline function (numflux::FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
                                    equations::AbstractShallowWaterEquations{NDIMS})
```

Reply:
Exactly. I just thought that if there are functions that apply to both SWE and SWEWD **and** which should live in Trixi.jl proper, such an additional abstraction might help.

Having said that, TBH, maybe I just do not properly understand the situation. IMHO, if it makes it less confusing if you add some duplication of FluxHLL here for the wet-dry equations, then do it: We do not expect the SWE implemention in Trixi.jl to grow much in the future, since virtually all SWE efforts should be focused on TrixiSW. Thus, these kind of redundancies are hopefully very limited.

Reply:
I think this a good idea and could help to reduce code duplication. However, I also think it might be not straight forward as we need to be careful that the abstraction is also compatible with all the sub-types (for example the `FluxHLL` with `AbstractShallowWaterEquations` would also affect `ShallowWaterEquationsQuasi1D`, which have a different variable ordering than the SWE) and probably also introduce a few changes in `Trixi.jl`.
As I expect the duplications to be limited to the SWEWD, I would suggest that we keep the current implementation for now, but to keep this option in mind I have opened an issue #24.